lock-api/src/main/java/com/palantir/lock/client/LockWatchTimeLockDiagnosticsLogger.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+public interface LockWatchTimeLockDiagnosticsLogger {
```

Comment:
This is similar to `LockWatchStarter` but used for a slightly different reason. There are two `LockWatchStarter` implementations: one calls TimeLock, and one just does internal bookkeeping (either in client or server side). `LockWatchTimeLockDiagnosticsLogger` only exists to avoid needing to stub tests etc etc. Can remove if we don't like this.

Reply:
nit: this interface makes sense. not a fan of the name though as this suggests it's some kind of logger and usually I'd expect `logState()` to be a cheap operation - it's not obvious that it's possibly remote! I'd suggest either renaming the method or the class so it's obvious this is the case (e.g. `logStateOnTimeLockServer();` or similar)